### PR TITLE
GEODE-8934: add incrby to supported commands

### PR DESCRIPTION
**Pull request by nonbinaryprogrammer**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/string/AbstractStringIntegrationTest.java [hunk 1]
```diff
@@ -26,141 +28,190 @@
...
+          jedis1.incrBy(key, increment);
+        },
+        (i) -> {
+          int increment = ThreadLocalRandom.current().nextInt(-50, 50);
+          expectedValue.addAndGet(increment);
+          jedis2.incrBy(key, increment);
+        }).run();
+
+    assertThat(Integer.parseInt(jedis1.get(key))).isEqualTo(expectedValue.get());
   }
```
Comment (sabbey37) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/string/AbstractStringIntegrationTest.java:216]: There's actually an `AbstractIncrByIntegrationTest` file. Some of these tests could be merged into that file. Others are duplicates. A test that isn't in either file: testing the max and min 64 bit signed integer value (9,223,372,036,854,775,807 and -9223372036854775808, respectively).  We should test both trying to increment a key that is set above/below these values (so setting a key to`9223372036854775808` then trying to incrby 1 (which results in `ERR value is not an integer or out of range`)) as well as incrementing a key that is right at the max/min value   (so setting a key to `9223372036854775807` then trying to incrby 1 (which results in `ERR increment or decrement would overflow`))

**Review by sabbey37 — CHANGES_REQUESTED** [review]

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/string/AbstractStringIntegrationTest.java (hunk 1 above)
Comment (nonbinaryprogrammer) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/string/AbstractStringIntegrationTest.java:216]: I just found that file as well. good idea to condense them into one. And I like that test so I'll definitely implement that

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/string/AbstractIncrByIntegrationTest.java [hunk 2]
```diff
@@ -18,79 +18,124 @@
...
+
+  @Test
+  public void testIncrBy_IncrementingMaxValueThrowsError() {
+    String key = "key";
+    Long increment = Long.MAX_VALUE / 2;
+
+    jedis1.set(key, String.valueOf(Long.MAX_VALUE));
+    assertThatThrownBy(() -> jedis1.incrBy(key, increment))
+        .hasMessageContaining("ERR increment or decrement would overflow");
+  }
```
Comment (sabbey37) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/string/AbstractIncrByIntegrationTest.java:117]: Thanks for adding this test and merging the files!  We get a different error when incrementing a key that is set above the max long value values (so setting a key to 9223372036854775808 then trying to incrby 1 (which results in `ERR value is not an integer or out of range`). Could we add a test for that as well?

**Review by sabbey37 — CHANGES_REQUESTED** [review]

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/string/AbstractStringIntegrationTest.java [hunk 3]
```diff
@@ -33,125 +33,128 @@
...
+
   private static final int REDIS_CLIENT_TIMEOUT =
       Math.toIntExact(GeodeAwaitility.getTimeout().toMillis());
 
   @Before
   public void setUp() {
-    jedis = new Jedis("localhost", getPort(), REDIS_CLIENT_TIMEOUT);
+    jedis1 = new Jedis("localhost", getPort(), REDIS_CLIENT_TIMEOUT);
+    jedis2 = new Jedis("localhost", getPort(), REDIS_CLIENT_TIMEOUT);
   }
```
Comment (sabbey37) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/string/AbstractStringIntegrationTest.java:46]: I should've mentioned this previously, but you could revert the changes in this file, and we don't need `jedis2` anymore.  You could always do that in a different PR though.

**Review by sabbey37 — APPROVED** [review]
